server/plugin/plg_backend_s3/index.go

```diff
@@ -418,7 +419,7 @@ func (s S3Backend) Touch(path string) error {
 	return err
 }
 
-func (s S3Backend) Save(path string, file io.Reader) error {
+func (s *S3Backend) Save(path string, file io.Reader) error {
```

Comment:
why making this a pointer receiver since we don't modify anything?

Reply:
Here is another consideration, if the `sync.Locker` is nested in the struct, there will be a warning of `passes lock by value`.
https://medium.com/codex/our-go-cache-library-choices-406f2662d6b

Reply:
Yes but I don't see anything that needs to be synchronised in here. Where did you see anything like this in the Save function?